src/datasources/db/utils.ts

```diff
@@ -0,0 +1,48 @@
+import { ICacheService } from '@/datasources/cache/cache.service.interface';
+import { CacheDir } from '@/datasources/cache/entities/cache-dir.entity';
+import { ILoggingService } from '@/logging/logging.interface';
+import { asError } from '@/logging/utils';
+import { InternalServerErrorException } from '@nestjs/common';
+import postgres from 'postgres';
+
+/**
+ * Returns the content from cache or executes the query and caches the result.
+ * If the specified {@link CacheDir} is empty, the query is executed and the result is cached.
+ * If the specified {@link CacheDir} is not empty, the pointed content is returned.
+ *
+ * @param loggingService {@link ILoggingService} to use for logging
+ * @param cacheService {@link ICacheService} to use for caching
+ * @param cacheDir {@link CacheDir} to use for caching
+ * @param query query to execute
+ * @param ttl time to live for the cache
+ * @returns content from cache or query result
+ */
+// TODO: add tests
```

Comment:
I'll add tests in a separate PR to avoid distracting from this PR's main purposes.